internal/core/analysis_api/handlers/validation.go

```diff
@@ -0,0 +1,84 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	resourceTypesProvider "github.com/panther-labs/panther/internal/compliance/snapshot_poller/models/aws"
+)
+
+// Populated by a call to refreshLogTypes
+var logtypeSetMap map[string]struct{}
+
+// Traverse a passed set of resource and return an error if any of them are not found in the current
+// list of valid resource types
+//
+// CAVEAT: This method uses a hardcoded list of existing resource types. If this method is returning
+// unexpected errors the hardcoded list is up to date.
+func ValidResourceTypeSet(checkResourceTypeSet []string) error {
+	for _, writeResourceTypeEntry := range checkResourceTypeSet {
+		if _, exists := resourceTypesProvider.ResourceTypes[writeResourceTypeEntry]; !exists {
+			// Found a resource type that doesnt exist
+			return errors.Errorf("%s", writeResourceTypeEntry)
+		}
+	}
+	return nil
+}
+
+// Request the logtypes-api for the current set of logtypes and assign the result list to 'logtypeSetMap'
+func refreshLogTypes() error {
+	// Temporary get log types for testing
+	logtypes, err := logtypesAPI.ListAvailableLogTypes(context.Background())
+	if err != nil {
+		return err
+	}
+	logtypeSetMap = make(map[string]struct{})
+	for _, logtype := range logtypes.LogTypes {
+		logtypeSetMap[logtype] = struct{}{}
+	}
+	return nil
+}
+
+// Return the existence of the passed logtype in the current logtypes.
+// NOTE: Accuret results require an updated logtypeSetMap - currently accomplished using the call to
+// 'refreshLogTypes'. That method makes a call to the log-types api, so use it as infrequently as possible
+// The refresh method can be called a single time for multiple individual log type validation checks.
+func logtypeIsValid(logtype string) (found bool) {
+	_, found = logtypeSetMap[logtype]
+	return
+}
+
+// Traverse a passed set of resource and return an error if any of them are not found in the current
+// list of valid resource types
+//
+// CAVEAT: This method will trigger a request to the log-types api EVERY time it is called.
+func validateLogtypeSet(logtypes []string) error {
+	if err := refreshLogTypes(); err != nil {
+		return err
+	}
+	for _, logtype := range logtypes {
+		if !logtypeIsValid(logtype) {
+			return errors.Errorf("%s", logtype)
+		}
+	}
+	return nil
+}
```

Comment:
This could be just:

```suggestion
func validateLogtypeSet(logtypes []string) error {
	availableLogTypes, err := logtypesAPI.ListAvailableLogTypes(context.TODO())
	if err != nil {
		return err
	}
	for _, logtype := range logtypes {
		if !stringset.Contains(availableLogTypes.LogTypes, logtype) {
			return errors.Errorf("%s", logtype)
		}
	}
	return nil
}
```

It achieves same functionality, it's also readable and saves few lines of code - we no longer need `refreshLogTypes` ,`logtypeIsValid` functions :)  

The important here is to get rid of the `var logtypeSetMap map[string]struct{}` package variable. We should try to keep package variables like that to a minimum: 
1. Any part of the package code can access them and mutate them. They are very useful sometimes (e.g. for caching values between Lambda invocations) but we should avoid them if possible since they add some code maintenance overhead since we need to make sure that e.g. no two goroutines try to modify them at the same time. 
2. They make unit testing challenging since unit tests running in parallel can modify them in parallel and lead to panics/failed tests.

Reply:
Also note the use of `context.TODO()` vs `context.Background()` used in the `refreshLogTypes()`. 

From `context.TODO()` docs: 
<pre>
// TODO returns a non-nil, empty Context. Code <b>should use context.TODO</b> when
// it's unclear which Context to use or <b>it is not yet available (because the
// surrounding function has not yet been extended to accept a Context
// parameter)</b>
</pre>

Reply:
Thanks for the context tip and I like the syntax code recommendation

Reply:
@kostaspap are you referring to the "github.com/jjeffery/stringset" package?

Reply:
We actually have our own `pkg/stringset`, but our implementation is a `[]string`, which is very odd, I don't know why we have that, to be honest. Seems very inefficient

My preference - stick to `map[string]struct{}` rather than importing an extra unnecessary package

Reply:
I could go both ways I like the syntax of @kostaspap and I very much like limiting external dependencies.

@austinbyers I put this together while I was waiting to hear what package to use, is this what you are thinking? I also have @kostaspap  suggestion working. 
```
	logtypeSetMap := make(map[string]struct{})
	for _, logtype := range availableLogTypes.LogTypes {
		logtypeSetMap[logtype] = struct{}{}
	}
	for _, lt := range logtypes {
		if _, found := logtypeSetMap[lt]; !found {
			return errors.Errorf("%s", lt)
		}
	}
```

Reply:
Yes, you've got the right idea. A local map in which you test inclusion 👍